app/javascript/packages/polyfill/index.js

```diff
@@ -37,6 +37,23 @@ const POLYFILLS = {
     },
     load: () => import(/* webpackChunkName: "custom-event-polyfill" */ 'custom-event-polyfill'),
   },
+  url: {
+    test() {
+      try {
+        // eslint-disable-next-line no-new
+        new URL('http://example.com');
+        // eslint-disable-next-line no-new
+        new URLSearchParams();
+        return true;
```

Comment:
what if we got fancy and made these more terse?

```suggestion
        // eslint-disable-next-line no-new
        return !!(new URL('http://example.com') && new URLSearchParams());
```

Reply:
> what if we got fancy and made these more terse?

Mm, it is more compact, but I think it's a bit more confusing to tease apart, since we don't really care so much what the values of the `URL` and `URLSearchParams` are, we just care that we're able to get past those invocations without an error being thrown.

If the issue is the ESLint directives, I don't feel like it's an especially valuable rule to begin with, so I'd personally be fine to disable it globally.

Or maybe we can go a different fancy path, in the spirit of [`_.attempt`](https://lodash.com/docs/4.17.15#attempt), something like...

```js
function try(fn) {
  try {
    fn();
    return true;
  } catch {
    return false;
  }
}

// ..

const POLYFILLS = {
  fetch: {
    test: () => 'fetch' in window,
    load: () => import(/* webpackChunkName: "whatwg-fetch" */ 'whatwg-fetch'),
  },
  classlist: {
    test: () => 'classList' in Element.prototype,
    load: () => import(/* webpackChunkName: "classlist-polyfill" */ 'classlist-polyfill'),
  },
  crypto: {
    test: () => 'crypto' in window,
    load: () => import(/* webpackChunkName: "webcrypto-shim" */ 'webcrypto-shim'),
  },
  'custom-event': {
    test: () => try(() => new window.CustomEvent('test')),
    load: () => import(/* webpackChunkName: "custom-event-polyfill" */ 'custom-event-polyfill'),
  },
  url: {
    test(): () => try(() => new URL('http://example.com')) && try(() => new URLSearchParams()),
    load: () =>
      import(
        /* webpackChunkName: "polyfill-library-url" */ 'polyfill-library/polyfills/__dist/URL/min'
      ),
  },
};
```

Reply:
nah, no need to redo things that much, I don't think it's that verbose now, fine to leave as-is

Reply:
I liked it 😄  (d41adfd)